File: apps/api/internal/models/menu.go
package models

import (
	"time"
)

type Menu struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	Title         string    `gorm:"size:160;not null" json:"title"`
	Description   string    `gorm:"type:text;not null" json:"description"`
	Category      string    `gorm:"size:80;not null;default:'lunch'" json:"category"`
	Price         int64     `gorm:"not null" json:"price"`
	AvailableDate time.Time `gorm:"type:date;not null;index" json:"available_date"`
	CutoffTime    time.Time `gorm:"not null" json:"cutoff_time"`
	MaxOrders     int       `gorm:"not null;default:0" json:"max_orders"`
	IsActive      bool      `gorm:"not null;default:true" json:"is_active"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: apps/web/src/types/menu.ts
export type Menu = {
  id: number;
  title: string;
  description: string;
  category: string;
  price: number;
  available_date: string;
  cutoff_time: string;
  max_orders: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
};

export type MenuPayload = {
  title: string;
  description: string;
  category: string;
  price: number;
  available_date: string;
  cutoff_time: string;
  max_orders: number;
  is_active: boolean;
};
